audio-worklet/design-pattern/lib/wa2.js

```diff
@@ -0,0 +1,289 @@
+/**
+ * @fileOverview WA2 library. A set of utilities to support WASM operations
+ * for the WebAudio.
+ */
+
+
+/**
+ * @namespace WA2
+ */
+var WA2 = {
+  // Basic byte unit of WASM heap. (16 bit = 2 bytes)
+  BYTES_PER_UNIT: Uint16Array.BYTES_PER_ELEMENT,
+
+  // Byte per audio sample. (32 bit float)
+  BYTES_PER_SAMPLE: Float32Array.BYTES_PER_ELEMENT,
+
+  // The max audio channel on Chrome is 32.
```

Comment:
Binary search on createBuffer to find the last value where it works. :-)

It would be best if we can avoid Chrome-specific magic numbers.  Or just change the comment to say "Minimum number of channels that all browsers must support".